cpp/src/gandiva/like_holder.cc

```diff
@@ -81,6 +81,12 @@ Status LikeHolder::Make(const FunctionNode& node, std::shared_ptr<LikeHolder>* h
       Status::Invalid(
           "'like' function requires a string literal as the second parameter"));
 
+  if (node.descriptor()->name() == "ilike") {
+    RE2::Options regex_op;
+    regex_op.set_case_sensitive(false);  // set case-insensitive for ilike function.
+
+    return Make(arrow::util::get<std::string>(literal->holder()), holder, regex_op);
```

Comment:
Why create a new method for this. Modify the existing one take options

Reply:
done